src/vmm/src/lib.rs

```diff
@@ -1005,6 +1006,44 @@ impl Vmm {
         Ok(entry_addr)
     }
 
+    /// Loads the initrd from a file into the given memory slice.
+    ///
+    /// * `vm_memory` - The guest memory the initrd is written to.
+    /// * `image` - The initrd image.
+    ///
+    /// Returns the result of initrd loading
+    fn load_initrd<F>(
+        vm_memory: &GuestMemory,
+        image: &mut F,
+    ) -> std::result::Result<InitrdInfo, LoadInitrdError>
+    where
+        F: Read + Seek,
+    {
+        use LoadInitrdError::*;
+
+        let size: usize;
+        // Get the image size
+        match image.seek(SeekFrom::End(0)) {
+            Ok(0) | Err(..) => return Err(ReadInitrd),
```

Comment:
We shouldn't ignore the error here. What we can do instead is return ReadInitrd(EINVAL) when seek returns 0 and the seek error on the Err(..) branch.